sdk/identity/src/token_credentials/powershell_credential.rs

```diff
@@ -0,0 +1,179 @@
+use azure_core::TokenResponse;
+use chrono::{DateTime, Utc};
+use oauth2::AccessToken;
+use regex::Regex;
+use serde::Deserialize;
+use std::io::Error;
+use std::str::Utf8Error;
+use std::{io::ErrorKind, path::PathBuf, str::FromStr};
+use tokio::process::Command;
+
+use super::TokenCredential;
+
+lazy_static::lazy_static! {
+    static ref POWERSHELL_PATH: PathBuf = get_default_powershell_path();
+}
+
+fn get_scope_resource(scope: &str) -> String {
+    Regex::new(r"/.default$")
+        .unwrap()
+        .replace(scope, "")
+        .to_string()
+}
+
+#[cfg(target_os = "windows")]
+fn get_default_powershell_path() -> PathBuf {
+    match Command::new("pwsh.exe").args("/?").output() {
+        Ok(_) => PathBuf::from_str("pwsh.exe"),
+        Err(_) => PathBuf::from_str("powershell.exe"),
+    }
+}
+
+#[cfg(not(target_os = "windows"))]
+fn get_default_powershell_path() -> PathBuf {
+    PathBuf::from_str("pwsh").unwrap()
+}
+
+#[derive(Debug, Clone, Deserialize)]
+#[serde(rename_all = "PascalCase")]
+struct PowerShellTokenResponse {
+    pub token: AccessToken,
+    pub expires_on: DateTime<Utc>,
+}
+
+/// Enables authenticating with Azure Active Directory using Azure PowerShell.
+///
+/// To use this credential:
+/// - Install the Azure PowerShell Module
+///   `Install-Module -Name Az -Scope CurrentUser -Repository PSGallery -Force`
+/// - Ensure that you are signed in using the `Connect-AzAccount` cmdlet.
+pub struct AzurePowerShellCredential {
+    powershell_path: Option<PathBuf>,
+}
+
+#[non_exhaustive]
+#[derive(Debug, thiserror::Error)]
+pub enum AzurePowerShellCredentialError {
+    #[error("Failed to import the Az.Account module (is it installed?): {0}")]
```

Comment:
I've done this in #306 - can you make the first letter of all the error messages lowercase? This is how the error handling working group recommends messages are formatted.